### PR TITLE
[NeuralChat] Support Neuralchat-TGI serving with Docker

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature
API not changed

## Description

Support Neuralchat and TGI serving with Docker. Service could be started using command "docker run".
JIRA ticket: https://jira.devtools.intel.com/browse/NLPTOOLKIU-1167

## Expected Behavior & Potential Risk

Service could be started using command "docker run".

## How has this PR been tested?

on SPR server

## Dependency Change?

no

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/docker/tgi_serving/tgi.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+host: 0.0.0.0
+port: 8000
+
+model_name_or_path: "Intel/neural-chat-7b-v3-1"
+device: "cpu"
+
+serving:
+    framework: "tgi"
+    tgi_engine_params:
+        model_id: "mistralai/Mistral-7B-Instruct-v0.1"
```
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/docker/tgi_serving/tgi.yaml:32]: Why need to set model_id while already configured model_name_or_path?

Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/docker/tgi_serving/tgi.yaml:32]: Will integrate into one model_name_or_path.

File: intel_extension_for_transformers/neural_chat/examples/serving/TGI/README.md [hunk 2]
```diff
@@ -63,7 +63,6 @@ You can customize the configuration file 'tgi.yaml' to match your environment se
 | model_name_or_path                | "./neural-chat-7b-v3-1"                 |
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/examples/serving/TGI/README.md:63]: better to make the readme more complete. After launching the service, how end users can interact with the service?

Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/examples/serving/TGI/README.md:63]: The example of comsumption of services are added to the README.

**Comment by hshen14** [comment]
Can we evaluate how to wrapper in a python API?

**Review by lvliang-intel — APPROVED** [review]

**Comment by letonghan** [comment]
> Can we evaluate how to wrapper in a python API?

TGI only offers the wrapped HTTP interface, and is implemented using RUST and Python. So it is costly to wrap TGI into python API.

**Pull request merged**